Content.Client/Box/BoxSystem.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Content.Client.Box.Components;
+using Content.Shared.Box;
+using Content.Shared.Box.Components;
+using Content.Shared.Interaction.Helpers;
+using Content.Shared.Movement.Components;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Box;
+
+public sealed class BoxSystem : SharedBoxSystem
+{
+    [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeNetworkEvent<PlayBoxEffectMessage>(OnBoxEffect);
+    }
+
+    private void OnBoxEffect(PlayBoxEffectMessage msg)
+    {
+        if (!TryComp<BoxComponent>(msg.Source, out var box))
+            return;
+
+        var xformQuery = GetEntityQuery<TransformComponent>();
+
+        if (!xformQuery.TryGetComponent(msg.Source, out var xform))
+            return;
+
+        //Any mob that can move should be surprised?
+        //God mind rework needs to come faster so it can just check for mind
+        //TODO: Replace with Mind Query when mind rework is in.
+        var mobMoverEntities = new HashSet<EntityUid>();
+        var mobMoverQuery = GetEntityQuery<MobMoverComponent>();
+
+        //Filter out entities in range to see that they're a mob and add them to the mobMoverEntities hash for faster lookup
+        foreach (var entity in _entityLookup.GetEntitiesInRange(xform.Coordinates, box.Distance, LookupFlags.Approximate))
+        {
+            if (!mobMoverQuery.HasComponent(entity) || msg.Mover == entity)
+                continue;
+
+            mobMoverEntities.Add(entity);
+        }
```

Comment:
There's now a `GetComponentsInRange` / `GetComponentsIntersecting` that should be used instead.

It basically does the same thing, except it automatically switches between this kind of check (is nearby && has comp) and the reverse (has comp && is nearby) depending on the number of entities with the requested component.